Tools/patchcheck/patchcheck.py

```diff
@@ -268,35 +250,28 @@ def ci(pull_request):
     file_paths = changed_files(base_branch)
     python_files = [fn for fn in file_paths if fn.endswith('.py')]
     c_files = [fn for fn in file_paths if fn.endswith(('.c', '.h'))]
-    doc_files = [fn for fn in file_paths if fn.startswith('Doc') and
-                 fn.endswith(('.rst', '.inc'))]
     fixed = []
     fixed.extend(normalize_whitespace(python_files))
     fixed.extend(normalize_c_whitespace(c_files))
-    fixed.extend(normalize_docs_whitespace(doc_files))
     if not fixed:
         print('No whitespace issues found')
     else:
-        print(f'Please fix the {len(fixed)} file(s) with whitespace issues')
-        print('(on UNIX you can run `make patchcheck` to make the fixes)')
+        count = len(fixed)
+        print(f'Please fix the {n_files_str(count)} with whitespace issues')
+        print('(on Unix you can run `make patchcheck` to make the fixes)')
         sys.exit(1)
 
+
 def main():
     base_branch = get_base_branch()
     file_paths = changed_files(base_branch)
     python_files = [fn for fn in file_paths if fn.endswith('.py')]
     c_files = [fn for fn in file_paths if fn.endswith(('.c', '.h'))]
-    doc_files = [fn for fn in file_paths if fn.startswith('Doc') and
-                 fn.endswith(('.rst', '.inc'))]
     misc_files = {p for p in file_paths if p.startswith('Misc')}
     # PEP 8 whitespace rules enforcement.
     normalize_whitespace(python_files)
     # C rules enforcement.
     normalize_c_whitespace(c_files)
-    # Doc whitespace enforcement.
-    normalize_docs_whitespace(doc_files)
-    # Docs updated.
-    docs_modified(doc_files)
```

Comment:
This will currently check if the user has written documentation for their patch (Docs updated... NO). If we delete from patchcheck, could we look into adding a check in pre-commit for if a user has added a NEWS entry or docs? 

We should also delete the function, if deleting the only usage.

Reply:
`docs_modified` doesn't really fit in pre-commit: it only reports and doesn't pass/fail.

And pre-commit hides output by default when it passing.

So I think I'll reinstate this for now, it's just a couple of lines.

Reply:
Perhaps we could modify the goal to be removal of ``patchcheck`` just from CI rather than entirely?

I think it does provide useful features to remind contributors (especially newer contributors) what a PR needs. (But also if there is a way to improve this, I'd be interested)

Reply:
I'd prefer it if we don't need to maintain the same linter/fixer in two places. 

I also wonder how many newer contributors run patchcheck, and for that matter how many older contributors too.

Anyway, once we move linters/fixers to pre-commit, if it's still useful to output a summary, then we can definitely keep that in some form or other.